api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -325,8 +333,9 @@ type KubeObjectProtectionStatus struct {
 type VolumeReplicationGroupStatus struct {
 	State State `json:"state,omitempty"`
 
-	// All the protected pvcs
+	// One of ProtectedPVCs or ProtectedCGs must be set
 	ProtectedPVCs []ProtectedPVC `json:"protectedPVCs,omitempty"`
+	ProtectedCGs  []ProtectedCG  `json:"protectedCGs,omitempty"`
```

Comment:
Similarly here we can just add a patch like so:

```
index 0332a027..d2362589 100644
--- a/api/v1alpha1/volumereplicationgroup_types.go
+++ b/api/v1alpha1/volumereplicationgroup_types.go
@@ -327,6 +327,7 @@ type VolumeReplicationGroupStatus struct {
 
        // All the protected pvcs
        ProtectedPVCs []ProtectedPVC `json:"protectedPVCs,omitempty"`
+       PVCGroups     []Groups       `json:"pvcgroups,omitempty"`
 
        // Conditions are the list of VRG's summary conditions and their status.
        Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Let ProtectedPVCs and their details remain as is, no need to disturb them.

We also will avoid needing to synthetically generate a name per group.

Reply:
According to Madhu, in case of groups we have no information on PVC level, only group have conditions. If so, why we need the whole ProtectedPVC list? It was my first thought, but then I decided, that ProtectedCG can "become" ProtectedPVC with list of pvc names in addition